### PR TITLE
VDS example

### LINKED ISSUE #74: Add an example on how to use the VDS on Data access page
ESA requested an example on how to use the verification dataset to do comparisons with other mission (included in the VDS)

### PR DESCRIPTION
**Pull request by skymandr**
Add information on the VDS API and a downloadable example to the data access page.

Resolves #74

### CODE REVIEW

**Comment by local-minimum** [comment]
Test error because they can't run two times at the same time?

**Comment by skymandr** [comment]
> Test error because they can't run two times at the same time?

I don't follow.

**Comment by local-minimum** [comment]
> > Test error because they can't run two times at the same time?
> 
> I don't follow.

You had one test failing and that fail looked kind of like it failed because both branch and pr-merge were running at the same time. Or did you have any other understanding of it? If you agree, then I suggest creating an issue unless we are promptly moving this testing away from jenkins.

**Comment by skymandr** [comment]
> > > Test error because they can't run two times at the same time?
> > 
> > 
> > I don't follow.
> 
> You had one test failing and that fail looked kind of like it failed because both branch and pr-merge were running at the same time. Or did you have any other understanding of it? If you agree, then I suggest creating an issue unless we are promptly moving this testing away from jenkins.

I created an issue, but @joakimmoller is already working on Github Actions, so it might soon be redundant.

File: src/examples/vds_example.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+
+# Setup the name space:
```
Comment (local-minimum) [inline src/examples/vds_example.py:3]: weird comment for importing dependencies

File: src/examples/vds_example.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import requests
+
+# Start by making a request to the root URI of the VDS API:
+r0 = requests.get("http://odin.rss.chalmers.se/rest_api/v4/vds/")
+
+# The request contains the returned JSON object, which in Python is a
+# dictionary.  This can be printed or inspected to find out its keys and
+# contents.  Let's assume that we have done that, or that we have read
```
Comment (local-minimum) [inline src/examples/vds_example.py:11]: multiple spaces in comment

File: src/examples/vds_example.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+# the API documentation, so that we know that 'FreqMode' is a key.
+# Use this to single out the frequency mode of interest, in this case 2:
+FM2 = [
+    x for x in r0.json()['VDS']
+    if x['FreqMode'] == 2
+][0]
```
Comment (local-minimum) [inline src/examples/vds_example.py:17]: it's not explained why 0th is used.

File: src/examples/vds_example.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+# Make a new request using the URI provided in the JSON object, and single
+# out the species O3 and the instrument MLS:
+r1 = requests.get(FM2["URL-collocation"])
+O3MLS = [
+    x for x in r1.json()['VDS']
+    if x['Species'] == 'O3' and x['Instrument'] == 'mls'
+][0]
```
Comment (local-minimum) [inline src/examples/vds_example.py:25]: same

File: src/examples/vds_example.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+# Repeat for the new object and a date of interest:
```
Comment (local-minimum) [inline src/examples/vds_example.py:27]: isn't really explained why this is a two step proceedure

File: src/examples/vds_example.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+r2 = requests.get(O3MLS['URL'])
+day = [
+    x for x in r2.json()['VDS']
+    if x['Date'] == '2012-09-15'
+][0]
```
Comment (local-minimum) [inline src/examples/vds_example.py:32]: same

File: src/examples/vds_example.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+# To get the detailed information about the collocated scans on that day,
+# we make one more request:
+r3 = requests.get(day['URL'])
+scanData = r3.json()['VDS']
+
+# scanData contains URIs for getting all the collocated Odin/SMR scans for
+# 2012-09-15, as well as the data from MLS.  As a final step, let's request
```
Comment (local-minimum) [inline src/examples/vds_example.py:40]:
```suggestion
# 2012-09-15, as well as the data from MLS. As a final step, let's request
```

File: src/examples/vds_example.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+# the latter for the first of the collocated scans for our chosen frequency
+# mode, species, instrument and day:
+r4 = requests.get(scanData[0]['URLS']['URL-mls-O3'])
```
Comment (local-minimum) [inline src/examples/vds_example.py:43]: the `-03` bit of this key isn't explained.

File: src/odinapi/templates/dataaccess.html [hunk 9]
```diff
@@ -357,6 +364,12 @@ <h2>Downloadable examples</h2>
 <a href="{{ url_for('static', filename='examples/get_l1b_for_period.py') }}">
   here</a>.
 </p>
+<p>
+The <em>Verification Data Set</em> API provides functionality for comparing
+Odin/SMR data with data from other instruments. An example Python program
```
Comment (local-minimum) [inline src/odinapi/templates/dataaccess.html:369]:
```suggestion
Odin/SMR data with data from other instruments. An example Python script
```

File: src/odinapi/templates/dataaccess.html [hunk 10]
```diff
@@ -357,6 +364,12 @@ <h2>Downloadable examples</h2>
...
+illustrating its use can be dowloaded
+<a href="{{ url_for('static', filename='examples/vds_example.py') }}">
+  here</a>.
```
Comment (local-minimum) [inline src/odinapi/templates/dataaccess.html:372]:
```suggestion
<a href="{{ url_for('static', filename='examples/vds_example.py') }}">here</a>.
```

linebreak of html at 80 doesn't make any sense. also multiple spaces but perhaps there's a need for an `&npsp;` between the link and "downloaded"?

File: src/examples/vds_example.py (hunk 3 above)
Comment (skymandr) [reply src/examples/vds_example.py:17]: Correct. I don't remember why...

File: src/examples/vds_example.py (hunk 5 above)
Comment (skymandr) [reply src/examples/vds_example.py:27]: That is only for clarity.

File: src/examples/vds_example.py (hunk 8 above)
Comment (skymandr) [reply src/examples/vds_example.py:43]: `O3` not `03`.

File: src/examples/vds_example.py (hunk 2 above)
Comment (skymandr) [reply src/examples/vds_example.py:11]: This is American standard, but I suppose we are all Brits...

File: src/examples/vds_example.py (hunk 5 above)
Comment (local-minimum) [reply src/examples/vds_example.py:27]: my point is that it's unclear to me what happens and why two steps are needed.

Comment (skymandr) [reply src/examples/vds_example.py:27]: Ah, you'd have to read the documentation for a detailed answer to that, but I'll add something to acknowledge it.

**Review by local-minimum — APPROVED** [review]

**Pull request merged**